### PR TITLE
Mds 3942 minespace submit incident notification

### PR DESCRIPTION
**Pull request by areyeslo**
# Main

- Provide the functionality to create new incidents on minespace. (The button to record a new incident has to be hidden when the code review will be done)

# Other
- Changes in styling were added on antd-overrides.scss and webpack.parts.js

# How to test
- Click on image
- Fill the required fields on the first-step form and click on Next: 
image
- Fill the required fields on the second-step form:  
image
- At the bottom of the form appears an alert with a legend to check understanding of contacting an inspector:
image
- When the user clicks on the checkbox, then the Submit button is enabled to create the new incident. 
image
- The newly created incident is added at the top of the Incidents table:
image

- Email notification example:
image

# Notes
https://bcmines.atlassian.net/browse/MDS-3942

### CODE REVIEW

File: services/core-api/app/api/constants.py [hunk 1]
```diff
@@ -163,5 +163,6 @@ def EXPLOSIVES_PERMIT_DOCUMENT_DOWNLOAD_TOKEN(token_guid):
 
 MAJOR_MINES_OFFICE_EMAIL = 'PermRecl@gov.bc.ca'
 VARIANCE_APPLICATION_EMAIL = 'kate.musgrove@gov.bc.ca'
+INCIDENTS_EMAIL = 'kate.musgrove@gov.bc.ca'
```
Comment (mmumby) [inline services/core-api/app/api/constants.py:167]: This should be: mineincidents@gov.bc.ca

File: services/core-api/app/api/incidents/models/mine_incident.py [hunk 2]
```diff
@@ -199,3 +202,14 @@ def validate_reported_timestamp(self, key, reported_timestamp):
             if reported_timestamp > datetime.datetime.utcnow():
                 raise AssertionError('reported_timestamp must not be in the future')
         return reported_timestamp
+
+    def send_incidents_email(self):
+        recipients = [INCIDENTS_EMAIL]
+
+        subject = f'Incident Notification for {self.mine_table.mine_name}'
+        body = f'<p>{self.mine_table.mine_name} (Mine no: {self.mine_table.mine_no}) has reported an incident in MineSpace.</p>'
```
Comment (mmumby) [inline services/core-api/app/api/incidents/models/mine_incident.py:211]: Let's include the incident type to the email as well. (above the "Incident information" )

File: services/core-api/app/api/mines/incidents/resources/mine_incidents.py [hunk 3]
```diff
@@ -182,6 +186,8 @@ def post(self, mine_guid):
 
         try:
             incident.save()
+            if is_minespace_user():
+                incident.send_incidents_email()
```
Comment (mmumby) [inline services/core-api/app/api/mines/incidents/resources/mine_incidents.py:196]: We'll have to use this check more throughout this API. Ie, proponents can only update specific fields, so we'll have to add more validation to the fields they cannot alter.

File: services/minespace-web/src/components/Forms/incidents/AddIncidentDetailForm.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  mineGuid: PropTypes.string.isRequired,
+  onFileLoad: PropTypes.func.isRequired,
+  onRemoveFile: PropTypes.func.isRequired,
+};
+
+const defaultProps = {};
+
+// eslint-disable-next-line react/prefer-stateless-function
+class AddIncidentDetailForm extends Component {
+  render() {
```
Comment (mmumby) [inline services/minespace-web/src/components/Forms/incidents/AddIncidentDetailForm.js]: Can you convert this to a stateless component?

File: services/minespace-web/src/components/dashboard/mine/incidents/Incidents.js [hunk 5]
```diff
@@ -24,18 +41,57 @@ export class Incidents extends Component {
...
       })
       .then(() => {
         this.setState({ isLoaded: true });
       });
   }
 
+  handleCreateIncident = (values) => {
+    return this.props.createMineIncident(this.props.mine.mine_guid, values).then(() => {
+      this.props.closeModal();
+      this.props.fetchIncidents({
```
Comment (mmumby) [inline services/minespace-web/src/components/dashboard/mine/incidents/Incidents.js]: Can you move the logic for "fetchIncidents" into a common method that both the 'componentDidmount' and this method can call? 

It also appears to be missing the loading state logic here.

File: services/minespace-web/src/constants/API.js [hunk 6]
```diff
@@ -22,6 +22,12 @@ export const VARIANCE_DOCUMENT = (mineGuid, varianceGuid, documentManagerGuid) =
 export const VARIANCE_STATUS_CODES = "/variances/status-codes";
 export const VARIANCE_DOCUMENT_CATEGORY_OPTIONS = "/variances/document-categories";
 
+// Incidents
```
Comment (mmumby) [inline services/minespace-web/src/constants/API.js]: These should already exist in the common folders.

**Review by mmumby — CHANGES_REQUESTED** [review]
Nice work, just a few small changes, and I'd like to see snapshot tests for all the new components.

File: services/core-api/app/api/constants.py (hunk 1 above)
Comment (areyeslo) [reply services/core-api/app/api/constants.py:167]: Changed. Thanks for the email.

File: services/core-api/app/api/incidents/models/mine_incident.py (hunk 2 above)
Comment (areyeslo) [reply services/core-api/app/api/incidents/models/mine_incident.py:211]: Added. Thanks for adding this detail.

File: services/minespace-web/src/components/Forms/incidents/AddIncidentDetailForm.js (hunk 4 above)
Comment (areyeslo) [reply services/minespace-web/src/components/Forms/incidents/AddIncidentDetailForm.js]: Converted :)

File: services/minespace-web/src/components/dashboard/mine/incidents/Incidents.js (hunk 5 above)
Comment (areyeslo) [reply services/minespace-web/src/components/dashboard/mine/incidents/Incidents.js]: Moved to common method.

File: services/minespace-web/src/constants/API.js (hunk 6 above)
Comment (areyeslo) [reply services/minespace-web/src/constants/API.js]: They exist in common folder. Deleted from here.

File: services/core-api/app/api/incidents/models/mine_incident.py (hunk 2 above)
Comment (mmumby) [reply services/core-api/app/api/incidents/models/mine_incident.py:211]: 'categories' is an array, you will have to map through and list the descriptions here.

Comment (areyeslo) [reply services/core-api/app/api/incidents/models/mine_incident.py:211]: Thanks, now returning the description :)

File: services/core-api/app/api/mines/incidents/resources/mine_incidents.py (hunk 3 above)
Comment (areyeslo) [reply services/core-api/app/api/mines/incidents/resources/mine_incidents.py:196]: Included validation through the POST method.

File: services/core-api/app/api/mines/incidents/resources/mine_incidents.py [hunk 7]
```diff
@@ -113,35 +119,38 @@ def post(self, mine_guid):
...
-            if not _compliance_article_is_do_subparagraph(sub):
-                raise BadRequest(
-                    'One of the provided compliance articles is not a sub-paragraph of section 1.7.3 (dangerous occurrences)'
-                )
-            incident.dangerous_occurrence_subparagraphs.append(sub)
+        if is_minespace_user() is not True:
+            incident.followup_inspection = data.get('followup_inspection')
+            incident.followup_inspection_date = data.get('followup_inspection_date')
+            incident.status_code = data.get('status_code')
+
```
Comment (mmumby) [inline services/core-api/app/api/mines/incidents/resources/mine_incidents.py:128]: 'status_code' should be getting set to 'PRE' if 'is_minespace_user'

Comment (areyeslo) [reply services/core-api/app/api/mines/incidents/resources/mine_incidents.py:128]: Done.

File: services/minespace-web/common/utils/helpers.js [hunk 8]
```diff
@@ -499,3 +499,12 @@ export const determineExemptionFeeStatus = (
   }
   return exemptionStatus;
 };
+
+export const formatBooleanToString = (value, defaultValue) => {
+  if (isNil(value)) {
+    return defaultValue;
+  } else {
+    response = value ? "Yes" : "No";
+  }
```
Comment (mmumby) [inline services/minespace-web/common/utils/helpers.js:509]: this is not how const are defined in JS. 


```
export const formatBooleanToString = (value, defaultValue) => {
let response;
  if (isNil(value)) {
   response = defaultValue;
  } else {
    response = value ? "Yes" : "No";
  }
  return response;
};

```

Comment (areyeslo) [reply services/minespace-web/common/utils/helpers.js:509]: My bad. Apologies for that.

**Review by mmumby — APPROVED** [review]
Well done, Arturo! 🔥

**Pull request merged**